internal/controllers/clusterrequest_controller_test.go

```diff
@@ -1562,9 +1586,58 @@ var _ = Describe("waitForNodePoolProvision", func() {
 		}
 	})
 
-	It("returns false when error fetching NodePool", func() {
-		rt := task.checkNodePoolProvisionStatus(ctx, np)
-		Expect(rt).To(Equal(false))
+	It("returns error when error fetching NodePool", func() {
+		provisioned, timedOutOrFailed, err := task.checkNodePoolProvisionStatus(ctx, np)
+		Expect(provisioned).To(Equal(false))
+		Expect(timedOutOrFailed).To(Equal(false))
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("returns failed when NodePool provisioning failed", func() {
+		provisionedCondition := metav1.Condition{
+			Type:   "Provisioned",
+			Status: metav1.ConditionFalse,
+			Reason: string(hwv1alpha1.Failed),
+		}
+		np.Status.Conditions = append(np.Status.Conditions, provisionedCondition)
+		Expect(c.Create(ctx, np)).To(Succeed())
+		provisioned, timedOutOrFailed, err := task.checkNodePoolProvisionStatus(ctx, np)
+		Expect(provisioned).To(Equal(false))
+		Expect(timedOutOrFailed).To(Equal(true)) // It should be failed
+		Expect(err).ToNot(HaveOccurred())
+		condition := meta.FindStatusCondition(cr.Status.Conditions, string(utils.CRconditionTypes.HardwareProvisioned))
+		Expect(condition).ToNot(BeNil())
+		Expect(condition.Status).To(Equal(metav1.ConditionFalse))
+		Expect(condition.Reason).To(Equal(string(hwv1alpha1.Failed)))
+	})
+
+	It("returns timeout when NodePool provisioning timed out", func() {
+		provisionedCondition := metav1.Condition{
+			Type:   "Provisioned",
+			Status: metav1.ConditionFalse,
+		}
+		np.Status.Conditions = append(np.Status.Conditions, provisionedCondition)
+		Expect(c.Create(ctx, np)).To(Succeed())
+
+		// First call to checkNodePoolProvisionStatus (before timeout)
+		provisioned, timedOutOrFailed, err := task.checkNodePoolProvisionStatus(ctx, np)
+		Expect(provisioned).To(Equal(false))
+		Expect(timedOutOrFailed).To(Equal(false))
+		Expect(err).ToNot(HaveOccurred())
+
+		// Simulate a delay to represent timeout (greater than 1 seconds)
+		time.Sleep(2 * time.Second) // This ensures the timeout occurs
```

Comment:
I think it's preferred to edit `reconciledCR.Status.NodePoolRef.HardwareProvisioningCheckStart` than wait here.

Reply:
It has been chanaged to move the start time back.